guard/src/commands/validate/generic_summary.rs

```diff
@@ -0,0 +1,175 @@
+       data=data_file,
+       rules=rules_file,
+       rule=info.rule,
+       path=info.path,
+       msg=info.message.replace("\n", ";"),
+    ))
+}
+
+fn unary_error_message(rules_file: &str, data_file: &str, op_msg: &str, info: &NameInfo<'_>) -> crate::rules::Result<String> {
+    Ok(format!("Property [{path}] in data [{data}] is not compliant with [{rules}/{rule}] because needed value at [{provided}] {op_msg}. Error Message [{msg}]",
+        path=info.path,
+        provided=info.provided.as_ref().map_or(&serde_json::Value::Null, std::convert::identity),
+        op_msg=op_msg,
+        data=data_file,
+        rules=rules_file,
+        rule=info.rule,
+        msg=info.message.replace("\n", ";"),
+    ))
+}
+
+fn binary_error_message(rules_file: &str, data_file: &str, op_msg: &str, info: &NameInfo<'_>) -> crate::rules::Result<String> {
+    Ok(format!("Property [{path}] in data [{data}] is not compliant with [{rules}/{rule}] because provided value [{provided}] {op_msg} match expected value [{expected}]. Error Message [{msg}]",
+               path=info.path,
+               provided=info.provided.as_ref().map_or(&serde_json::Value::Null, std::convert::identity),
+               op_msg=op_msg,
+               data=data_file,
+               rules=rules_file,
+               rule=info.rule,
+               msg=info.message.replace("\n", ";"),
+               expected=info.expected.as_ref().map_or(&serde_json::Value::Null, |v| v)
+    ))
+}
+
+impl GenericReporter for SingleLineSummary {
+    fn report(&self,
+              writer: &mut dyn Write,
+              rules_file_name: &str,
+              data_file_name: &str,
+              failed: HashMap<String, Vec<NameInfo<'_>>>,
+              passed: HashSet<String>,
+              skipped: HashSet<String>, longest_rule_len: usize) -> crate::rules::Result<()>
+    {
+        writeln!(writer, "Evaluation of rules {} against data {}", rules_file_name, data_file_name)?;
+        if !failed.is_empty() {
+            writeln!(writer, "--");
+        }
+        for (_rule, clauses) in failed {
+            super::common::print_name_info(
+                writer,
+                &clauses,
+                longest_rule_len,
+                rules_file_name,
+                data_file_name,
+                retrieval_error_message,
+                unary_error_message,
+                binary_error_message
+            )?;
+        }
+        if !passed.is_empty() {
+            writeln!(writer, "--");
+        }
+        for pass in passed {
+            writeln!(writer, "Rule [{}/{}] is compliant for data [{}]", rules_file_name, pass, data_file_name);
+        }
+
+        if !skipped.is_empty() {
+            writeln!(writer, "--");
+        }
+        for skip in skipped {
+            writeln!(writer, "Rule [{}/{}] is not applicable for data [{}]", rules_file_name, skip, data_file_name);
+        }
+        writeln!(writer, "--");
+        Ok(())
```

Comment:
This is duplicated in both cfn reporter and here, any way we can pull this out into a common construct? Not a 100% sure what the rust best practices are, but something like an abstract class maybe?

Reply:
This was deliberate, as messaging could potentially differ in the future for these reporters.

Reply:
Can we use string constants for capturing the boiler plate text for the error messages?